src/java_bytecode/java_utils.h

```diff
@@ -64,4 +65,9 @@ irep_idt resolve_friendly_method_name(
   const symbol_tablet &symbol_table,
   std::string &error);
 
+/// Dereference an expression and flag it for a null-pointer check
+/// \param expr: expression to dereference and check
+/// \param type: expected result type (typically expr.type().subtype())
```

Comment:
Seems to be normal for that header. Personally I prefer the docs in the header, but we should move or keep en masse.